src/vacation_calendar/chore.py

```diff
@@ -0,0 +1,185 @@
+AP_VAC = 25.5  # from config) (VAC anno precedente)
+
+
+class TimeOffType(StrEnum):
+    VAC = "VACATION"
+    ROL = "PTO"  # Paid Time Off
+    VAC_MAT = "VAC_MAT"
+    ROL_MAT = "PTO_MAT"
+    VAC_RES = "VAC_RES"
+    ROL_RES = "PTO_RES"
+
+
+def read_config():
+    # NOTE: this will be probably removed using pydantic
+    if not exists(CONFIG_PATH):
+        d = {}
+    else:
+        with open(CONFIG_PATH, "r") as f:
+            d = json.load(f)
+    return d
+
+
+def create_base_calendar() -> pd.DataFrame:
+    """Generate the base calendar, with holidays and weekends corresponding to the config files
+    Returns:
+        (pd.DataFrame): the calendar dataframe
+    """
+    year_holidays = holidays.country_holidays(country, years=current_year, subdiv=subdiv)
+    start = datetime.datetime(current_year, month=1, day=1)
+    end = datetime.datetime(current_year, month=12, day=31)
+    dates = pd.date_range(start=start, end=end, freq="D")
+    dates_df = dates.to_frame(index=False, name="DATE")
+    dates_df["MONTH"] = dates_df.DATE.apply(lambda ser: ser.month)
+    dates_df["YEAR"] = dates_df.DATE.apply(lambda ser: ser.year)
+    dates_df["IS_WEEKEND"] = dates_df["DATE"].dt.dayofweek.isin([5, 6])
+    dates_df["HOLIDAY"] = (
+        dates_df["DATE"]
+        .apply(
+            lambda day: year_holidays[day] if day in year_holidays else pd.NA,
+        )
+        .astype("category")
+    )
+    dates_df["IS_HOLIDAY"] = np.where(dates_df["HOLIDAY"].notna(), True, False)
+    dates_df[TimeOffType.VAC] = 0.0
+    dates_df[TimeOffType.ROL] = 0.0
+    return dates_df
+
+
+@singledispatch
+def add_to_calendar(
+    date: Union[datetime.date, str], df: pd.DataFrame, value: float, ttype: TimeOffType
+) -> pd.DataFrame:
+    """
+    Add the input value to the calendar, at the passed date.
+    Args:
+        date (Union[datetime.date, str]): date in which the time off will be taken
+        df (pd.DataFrame): input calendar df
+        value (float): amount of time off hours. Canno be higher than the `WORKING_HOURS`
+        ttype (TimeOffType): the type of time off; ROL for PaidTimeOff, VAC for Vacation.
+    Returns:
+        (pd.DataFrame): the df with the added time off
+    Raises:
+        ValueError if:
+        - `value` is higher than the `WORKING_HOURS`
+        - the chosen date is not present in the calendar
+        - the chosen date is a weekend or a holiday
+    """
+    if value > WORKING_HOURS:
+        raise ValueError("The amount per day cannot be higher than the working hours.")
+
+    mask = df.DATE == date
+
+    if df[mask].empty:
+        raise ValueError(f"The chosen date '{date}' does not exist in the calendar")
+
+    if (df[mask][["IS_WEEKEND", "IS_HOLIDAY"]]).any().any():
+        raise ValueError("The chosen day is a weekend or holiday")
+
+    df.loc[mask, ttype] = value
+    return df
+
+
+@add_to_calendar.register
+def add_range_to_calendar(
+    date: list, df: pd.DataFrame, value: float, ttype: TimeOffType
```

Comment:
Since we specify and use the date a single range a more strict typing such as `tuple[datetime.date]` could be better.
```suggestion
    date: tuple[datetime.date, datetime.date], df: pd.DataFrame, value: float, ttype: TimeOffType
```